### DISCUSSION TITLE
multi-tenancy with subdomain using the same project directory.

### DISCUSSION THREAD

**Discussion by BibinVJ** [leokhoa/laragon#630] [Q&A] [answered]
Hi!,

I am using tenancy for laravel and with the same project directory code I need to achieve multitenancy.
my project is in www/project

when I go to project.test     (it works fine)
what I want to achieve is even if I go to, 
- user.project.test
- user2.project.test

It all should direct me to www/project

I tried adding  *.project.test  in hosts file
still not working, anyone got any help

**Comment by Pen-y-Fan** [comment]
See this question on Configuring subdomain in Laragon #584 

Once the subdomain is defined Apache can point to the same project, you do not need to create multiple directories.  

The project will need middleware to detect the subdomain.

**Reply by BibinVJ** [reply]
it is a package and is working fine when i use xampp,
but here it is giving dns not found error!

**Reply by BibinVJ** [reply]
yes! now i have found the issue

when i manually add the,
- user.project.test 
to windows/hosts file it is working

but when i use  *.project.test , there it is not!

got any idea how to solve it

**Reply by aliharoondev** [reply]
Did you find any solution for that?

**Reply by BibinVJ** [reply]
No not yet, I had to manually add my subdomain in my host file!

**Reply by Satal** [reply]
I could be mistaken but I don't think the windows/hosts file accepts wildcards. I think you need to manually add each one. I have seen something previously about setting up a local DNS server that will handle this but it felt like a bit much for me.

**Reply by BibinVJ** [reply]
But it accepts wildcard for localhost.
User1.localhost
User2.localhost
Works very fine!

Yeah maybe it is not supported for other domain names. But, how could that be, if it supports wildcard for localhost domain. It should be workable for all right?

**Reply by enniosousa** [reply, marked as answer]
Laragon edit the hosts (`C:\WINDOWS\system32\drivers\etc\hosts`) to point the domain to `127.0.0.1`

So each subdomain you need add to windows hosts manually because it does not supports wildcard.

Google Chrome always resolves `localhost` and `*.localhost` and `*.*.localhost` ... to 127.0.0.1 without check windows hosts

So instead use `{name}.test` start use `{name}.localhost`

![image](https://github.com/leokhoa/laragon/assets/8731370/ec6003b4-b201-45d6-9f37-027c5bc0cab8)

**Reply by BibinVJ** [reply]
Yes!

Hope, laragon finds a way around this :)

**Reply by leokhoa** [reply]
Thanks @enniosousa 
Next version of Laragon, we will use .localhost by default so Laragon doesn't need to mess with the hosts file.

**Reply by luisAntonioLAGS** [reply]
@leokhoa, When will the new version come out?

**Reply by SyntaxByte-Solution** [reply]
@leokhoa 
When is the next Laragon coming out please?

Please try to handle subdomain creations

**Reply by helderlj** [reply]
> Thanks @enniosousa Next version of Laragon, we will use .localhost by default so Laragon doesn't need to mess with the hosts file.

unfortunately that was not the case with 8.0
